File: backend/src/Sgr/V1/ReciboBundle/Service/UtilitariosService.php
<?php

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

namespace App\Sgr\V1\ReciboBundle\Service;

/**
 * Description of UserService
 *
 * @author Odenilton
 */
use App\Sgr\BaseBundle\Service\BaseService;
use App\Sgr\BaseBundle\Exceptions\NegocioException;
use App\Sgr\BaseBundle\Utils\Configurator;
use App\Sgr\V1\ReciboBundle\Repository\UtilitariosRepository;
use Symfony\Component\DependencyInjection\ContainerInterface;

class UtilitariosService extends BaseService {

    public function __construct(ContainerInterface $container = null) {
        parent::__construct($container);
        $this->entity = "";
        $this->repository = new UtilitariosRepository();
    }

    public function obterAnos() {
        return $this->repository->obterAnos();
    }

    public function obterMeses() {
        return $this->repository->obterMeses();
    }

    public function obterOrgaos() {
        return $this->entityManager->getRepository('ReciboBundle:Orgao')->findAll();
    }

    public function obterEstados() {
        return $this->repository->obterEstados();
    }
    
    public function obterBancos() {
        return $this->repository->obterBancos();
    }

    public function obterRecibo($id) {
        return $this->entityManager->getRepository('ReciboBundle:Recibo')->findOneById($id);
    }

}

File: backend-java/src/main/java/br/com/oljsoft/sgr/service/UtilitariosService.java
package br.com.oljsoft.sgr.service;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import br.com.oljsoft.sgr.model.Orgao;
import br.com.oljsoft.sgr.repository.UtilitariosRepository;
import br.com.oljsoft.sgr.utils.dto.BancoDTO;
import br.com.oljsoft.sgr.utils.dto.EstadoDTO;
import br.com.oljsoft.sgr.utils.dto.ObjetoEnumDTO;

@Service
public class UtilitariosService {

	@Autowired
	private UtilitariosRepository repository;

	public Optional<List<EstadoDTO>> obterEstados() {
		return Optional.ofNullable(this.repository.obterEstados());
	}
	
	public Optional<List<ObjetoEnumDTO>> obterMeses() {
		return Optional.ofNullable(this.repository.obterMeses());
	}

	public Optional<List<ObjetoEnumDTO>> obterAnos() {
		return Optional.ofNullable(this.repository.obterAnos());
	}
	
	public Optional<List<Orgao>> obterOrgaos() {
		return Optional.ofNullable(this.repository.obterOrgaos());
	}

	public Optional<List<BancoDTO>> obterBancos() {
		return Optional.ofNullable(this.repository.obterBancos());
	}
	
}
